tests/components/speedtestdotnet/test_sensor.py

```diff
@@ -0,0 +1,47 @@
+"""Tests for SpeedTest sensors."""
+from homeassistant.components import speedtestdotnet
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.speedtestdotnet.const import DEFAULT_NAME, SENSOR_TYPES
+
+from . import MOCK_RESULTS, MOCK_SERVER_LIST
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+async def test_speedtestdotnet_sensors(hass):
+    """Test sensors created for speedtestdotnet integration."""
+    entry = MockConfigEntry(domain=speedtestdotnet.DOMAIN, data={})
+    entry.add_to_hass(hass)
+
+    with patch("speedtest.Speedtest") as mock_api:
+        mock_api.return_value.get_best_server.return_value = MOCK_SERVER_LIST[
+            "Server1"
+        ][0]
+        mock_api.return_value.results.dict.return_value = MOCK_RESULTS
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 3
+        print(hass.states.async_entity_ids(SENSOR_DOMAIN))
+
+        await hass.data[speedtestdotnet.DOMAIN].async_update()
```

Comment:
Don't call integration details in the tests. Take control of time and move it forward to force an update if needed. Patch and fire a time changed event with the helper in tests/common.py.